xapian-core/configure.ac

```diff
@@ -273,6 +273,17 @@ AC_PREPROC_IFELSE([AC_LANG_SOURCE([[
 #endif
 ]])], [win32=no], [win32=yes])
 
+AC_PREPROC_IFELSE([AC_LANG_SOURCE([[
+#if defined __EMSCRIPTEN__
+#error EMSCRIPTEN
+#endif
+]])], [emscripten=no], [emscripten=yes])
+
+if test "$emscripten" = yes ; then
+  AM_CXXFLAGS="$AM_CXXFLAGS -Oz -s USE_ZLIB=1"
```

Comment:
I don't think this will work as intended - compilation commands use `$(AM_CXXFLAGS) $(CXXFLAGS)` and `CXXFLAGS` defaults to `-O2 -g` with GCC, so that results in `... -Oz ... -O2 ...` which will mean `-Oz` will be effectively be ignored.

I notice the README instructions and the CI build both explicitly specify the same CXXFLAGS, presumably because of this.

I wonder if it's simpler to skip doing anything specific to emscripten in configure - the standard `./configure && make` won't work anyway since you need to use the `emconfigure` and `emmake` wrappers, and there are other required options specified by the commands in the README and CI as well.

Reply:
Followed your suggestion and removed the emscripten specifics from `configure.ac`.